certbot/main.py

```diff
@@ -600,25 +593,29 @@ def run(config, plugins):  # pylint: disable=too-many-branches,too-many-locals
     except errors.PluginSelectionError as e:
         return e.message
 
-    domains, certname = _find_domains_or_certname(config, installer)
-
     # TODO: Handle errors from _init_le_client?
     le_client = _init_le_client(config, authenticator, installer)
 
-    action, lineage = _auth_from_available(le_client, config, domains, certname)
+    domains, certname = _find_domains_or_certname(config, installer)
+    should_get_cert, lineage = _find_cert(config, domains, certname)
+
+    new_lineage = lineage
+    if should_get_cert:
+        new_lineage = _get_and_save_cert(le_client, config, domains,
+            certname, lineage)
 
-    le_client.deploy_certificate(
-        domains, lineage.privkey, lineage.cert,
-        lineage.chain, lineage.fullchain)
+    cert_path = new_lineage.cert_path if new_lineage else None
```

Comment:
`new_lineage` cannot be `False`, because then we would have already errored out. If `new_lineage` is `None`, it means we've done a dry run. `new_lineage` should not be `None` unless we've done a dry run.